### ISSUE TITLE
Need double TAB to get focus

### ISSUE THREAD

**Issue by andriasmelianus** [iliyaZelenko/vue-cool-select#98]
Hi, thanks for this really nice component!
I found a little unexpected behavior here when I want to focus to this component from previous input control with `Tab` key. I need to stroke `Tab` key twice to focus and start typing on this component.
![ezgif-1-27bc62878350](https://user-images.githubusercontent.com/7449543/55211777-123e1380-5220-11e9-860b-0d10cf5ba835.gif)
I have tested it on Chromium 73 and Firefox 66 on Ubuntu 16.04.
Thanks.

**Comment by iliyaZelenko** [comment]
Hi, thank you very much for finding a bug, today a new version will be released where this will be fixed. I'll let you know when this happens.

I see your fields may have a green color. If there are no errors, I can do this for you.

**Comment by iliyaZelenko** [comment]
I think I fixed the problem with the navigation through the <kbd>Tab</kbd>, but I need your confirmation.

I also added a new `successful` option that adds success styles. I also added 4 new slots (`input-before`, `input-start`, `input-end`, `input-after`) that allow you to place your content in different parts of the component. Using slot `input-end`, you can put a tick as you have in the screenshot.

I wrote a sample code that works for me (note: the new slot syntax is used):

```vue
<cool-select
  v-model="selected"
  :items="items"
  :error-message="errorMessage"
  :successful="!!(!errorMessage && selected)"
  placeholder="Select name"
>
  <template
    v-if="!errorMessage && selected"
    #input-end
  >
    <span style="color: green; margin-right: 8px;">
      ✔
    </span>
  </template>
</cool-select>
```

Waiting for your reply.

**Comment by iliyaZelenko** [comment]
@andriasmelianus I released a hotfix (`2.10.1`) where I fixed a small bug, make sure you install this version.

You can also write your suggestions for improving this component, I will consider them.

**Comment by andriasmelianus** [comment]
Hi @iliyaZelenko thank you so much for the extra effort & great features!
But I will try those later, for now, lets focus on this little bug first related to the <kbd>Tab</kbd> key.

I have install `2.10.1`, but the component refuse to focus and skipping to the next input control instead.
Here is the recorded behavior when I stroke <kbd>Tab</kbd> and <kbd>Shift</kbd> + <kbd>Tab</kbd>
![ezgif-4-e436b2e15298](https://user-images.githubusercontent.com/7449543/55243312-8ce34f00-5271-11e9-8cc5-1dcd239e2857.gif)

I also have checked this:
> [Demo on Codesanbox.](https://codesandbox.io/s/4zlkjr9xow?fontsize=14&initialpath=%2Fvalidation&module=%2Fsrc%2Fexamples%2FValidation.vue)

It works fine there and I realize, that this problem occurs when we combine the component with native or other input control (e.g. `<b-form-input>` from BootstrapVue) in the same form.

I hope you have the solution or alternative ways to solve this problem.

**Comment by iliyaZelenko** [comment]
This seems to be completely fixed in version `2.10.2`. I created for you a [demonstration project](https://codesandbox.io/s/zl8753lp34?fontsize=14&module=%2Fsrc%2FApp.vue) where you can see this.

Thank you for finding this error and sorry for seeing it in your project.

**Comment by andriasmelianus** [comment]
That's awesome! :handshake: :clap:
The component is working perfectly now.
:top: Lets see this repo crawls to the top soon! :smiley:
